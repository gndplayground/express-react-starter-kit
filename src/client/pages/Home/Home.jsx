/* global __ENV */

import React from 'react';
import {fetchAPI} from '../../utilities/api/fetchAPI';
import {APIUnauthorized, APIInternalError} from '../../utilities/api/apiErrors';
import {connect} from 'react-redux';

const mapStateToProps = (state) => {

    let user = state.user;

    return {
        token: user.token,
    }
};

class Home extends React.Component {


    constructor(props) {
        super(props);

        this.state = {
            checkAPI: {}
        }
    }

    checkAPI = () => {

        fetchAPI(`${__ENV.API_LOCATION}/dev/check`, {
            mode: 'cors',
            method: 'POST',
            headers: {
                Authorization: 'Bearer ' + this.props.token
            },
        }).then((data)=> {
            this.setState({
                checkAPI: {
                    ok: true,
                    response: data.user,
                }
            })
        }).catch((e) => {
            this.setState({
                checkAPI: {
                    ok: false,
                    response: e,
                }
            })
        })

    };

    render() {


        const apiResponse = this.state.checkAPI.response;

        let checkStatus = '';

        if (this.state.checkAPI.response) {


            if (!this.state.checkAPI.ok) {

                if (apiResponse instanceof APIUnauthorized) {
                    checkStatus = '401 - No authorization token was found in header or token is invalid';
                }

                if (apiResponse instanceof APIInternalError) {
                    checkStatus = '500 -There is a problem in server. Please check your config';
                }
            }
            else {
                checkStatus = (
                    <span>Access private api success with username: ${this.state.checkAPI.response.email}
                    <br/>Token: ${this.props.token}</span>
                )
            }
        }

        return (
            <div>
                <h2 className="route-title">Home page</h2>
                <p>This is our hom page.</p>
                <br/>
                <p>Please login then try to check access to private api</p>
                <p>{`${__ENV.API_LOCATION}/dev/check`}</p>
                <button onClick={this.checkAPI}>Check access</button>
                <br/>
                <p>{checkStatus}</p>
            </div>
        )
    }
}

Home.propTypes = {
    token: React.PropTypes.string.isRequired,
    children: React.PropTypes.node
};

export default connect(
    mapStateToProps,
    null,
)(Home)