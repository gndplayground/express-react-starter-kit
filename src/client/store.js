import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';
import auth from './auth';

import reducers from './reducers';

export const sagaMiddleware = createSagaMiddleware();

export const store = createStore(
    reducers,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
    compose(
        applyMiddleware(sagaMiddleware)
    )
);

// Read local storage. If we have user data we parse it and update redux state
try {

    if (window.localStorage.hasOwnProperty('user')) {
        store.dispatch(auth.actions.userSet(JSON.parse(window.localStorage.user)))
    }
}
catch (e) {

    window.localStorage.clear();
    console.log('Something wrong with local storage. Cleared local storage');
}