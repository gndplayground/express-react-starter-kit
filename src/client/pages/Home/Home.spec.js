import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Home from './Home';
import {store} from '../../store';



test('Should page home render correctly', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Home/>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});