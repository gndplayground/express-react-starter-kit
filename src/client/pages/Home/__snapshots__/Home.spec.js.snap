exports[`test Should page home render correctly 1`] = `
<div>
  <h2
    className="route-title">
    Home page
  </h2>
  <p>
    This is our hom page.
  </p>
  <br />
  <p>
    Please login then try to check access to private api
  </p>
  <p>
    http://localhost:3031/dev/check
  </p>
  <button
    onClick={[Function]}>
    Check access
  </button>
  <br />
  <p>
    
  </p>
</div>
`;
